src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post from which the image will be deleted.
+ *       - in: path
+ *         name: imgID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the image to be deleted from Cloudinary.
+ *     responses:
+ *       200:
+ *         description: Success message indicating that the image was deleted from both the post and Cloudinary.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Status message confirming the successful deletion of the image.
+ *                 error:
+ *                   type: object
+ *                   description: Contains error details if an error occurred during image deletion from Cloudinary.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred.
+ *                 error:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.post('/singleImgDel/:postID/:imgID', auth, deleteSinglePostImage);
+
+/**
+ * @swagger
+ * /cancel-delete:
+ *   post:
+ *     tags:
+ *       - General operations
+ *     summary: Handle cancel delete operation, Must be connected as user
+ *     description: Endpoint to handle the cancellation of a delete operation. Returns a success message indicating that the delete operation was cancelled.
+ *     responses:
+ *       200:
+ *         description: Success message confirming that the delete operation was cancelled.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Status message confirming the cancellation of the delete operation.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred.
+ *                 err:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.post('/onCancelImgDel', auth, onCancelDelete);
 
+/**
+ * @swagger
+ * /posts/{postID}:
+ *   put:
+ *     tags:
+ *       - Posts operations
+ *     summary: Update a post ,Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"